### ISSUE TITLE
Bug: Wrong "invalidFields" order with dynamic fields

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/react-advanced-form#73]
## What
The array of invalid fields (`invalidFields`) provided as an argument property of `onInvalid` callback seems to have messed up order of inputs when there are some dynamic fields appearing during the workflow with the form.

## Why
At first, my guess was that the `fields` reference taken to compose the array of invalid fields is taken at the wrong (past) moment, however, I see that it's taken immediately within the callback itself, so it should be up-to-date.

## Steps to reproduce
```jsx
<Form>
  <Field.Input name="i1" required />

  <Condition when={...}>
    <Field.Input name="i2" required />
  </Condition>

  <Field.Input name="i3" required />
</Form>
```
Once `<Condition>` is false, the order of `invalidFields` is correct (`[{ i1 }, { i2 }, { i3 }]`).

However, once you provide `i1` value, and condition resolves with empty field, then the order is i3, i2.

**Comment by kettanaito** [comment]
## TL;DR
The order of fields in `Form.state.fields` is dictated by the time they have registered, not by their order in the actual DOM.

## Update
The reason for this behavior is the order of the newly registered fields in the `Form.state`.

Once condition resolves, its children get rendered. If there are any fields, they will fire `registerWith` just as they should, registering themselves in the Form's state. However, they are added to the very end of the `Form.state.fields`, due to the order of they registration.

Considering this, I'm not sure how to deal with this behavior...

## Possible solutions
1. Ensure fields are sorted in the `Form.state.fields` based on their `fieldProps.ref.clientRect.top` property. Hackish and unreliable.
2. Make `invalidFields` unsorted from the beginning. Reliable and makes sense, but deprives the end developer of reliability of the ordered invalid fields (he would need to compare `ClientRect.top` coordinates himself to define proper order).

**Comment by kettanaito** [comment]
Closing for now, as there's no appropriate solution with the current setup.
